[SYCL] Assignment operator tests for 0-dim buffer accessor and local_accessor

andylshort opened the pull request:
SYCL2020 defines assignment operators for 0-dimensional accessors. This patch adds tests for the buffer `accessor` and `local_accessor` classes, with the `host_accessor` test being done as unit test in the intel/llvm repo instead.

Commit:
Assignment operator E2E tests for buffer accessor and local_accessor

andylshort commented:
/verify with https://github.com/intel/llvm/pull/8069

Commit:
Adjusted how accessor is created in test

Same things, tests pass, formatting rules followed.

Commit:
clang-format-related formatting fixes.

andylshort commented:
/verify with https://github.com/intel/llvm/pull/8069

Commit:
Merge branch 'intel' into alamzeds/0-dim-accessor-assignment-op-tests

Commit:
Merge branch 'intel' into alamzeds/0-dim-accessor-assignment-op-tests

andylshort commented:
The implementation half of this has been merged as part of https://github.com/intel/llvm/pull/8069

AlexeySachkov commented on SYCL/Basic/accessor/accessor.cpp:
```diff
@@ -1051,5 +1051,36 @@ int main() {
+      LocalAccT LocalAcc(CGH);
+      CGH.single_task<class local_acc_0_dim_assignment>([=]() {
+        LocalAcc = 64;
+        assert(LocalAcc == 64);
```
SYCL spec does not guarantee that `assert` will work in a kernel. Therefore, I would suggest to storing `LocalAcc` value into some other non-local accessor to check it on host

andylshort replied:
I should have now addressed this in https://github.com/intel/llvm-test-suite/pull/1570/commits/fe9cf992aa8236b5d4dfa8dc8790d0314f4bc928

Commit:
Merge branch 'intel' into alamzeds/0-dim-accessor-assignment-op-tests

Commit:
Fix for accessor test on GPUs

Commit:
Fixed local_accessor test

assert() isn't guaranteed to work in a kernel, so assigned to an accessor instead.

Got rid of pointless aliases for these two tests also.

steffenlarsen commented:
/verify with https://github.com/intel/llvm/pull/8069

steffenlarsen commented:
Automatic CI is not running at the moment, so lets try verifying with the merged PR.

AlexeySachkov commented:
> Automatic CI is not running at the moment, so lets try verifying with the merged PR.

Regular CI finished and it seems like the whole Linux run has failed due to some environment issue. However, `verify with` check passed, so I suggest that we merge this PR. @steffenlarsen, any objections?

steffenlarsen commented:
The Linux failures should be addressed with next nightly build, but since verify is unaffected I agree with going ahead with a merge.
